### ISSUE TITLE
Feature Request: Config setting to quit on server disconnect

### ISSUE THREAD

**Issue by alasdairdc** [dbcli/mycli#162]
I realise you prompt to reconnect to server when its gone away (#26)

Would it be possible to have a config setting to force quit the client when the server idle timeout has been reached and it has 'gone away'?

**Comment by amjith** [comment]
Can you give me a use case for this? I'd like to keep the number of configurable options to just the essentials. So if you can give me a good enough reason to do this, I'll be happy to oblige.

**Comment by alasdairdc** [comment]
I understand the need to keep config options to a minimal and consistent with the builtin cli for mysql.  The use case that irks me most as a SysAdmin is:  I have a bash timeout set which works (and i realise any user could use any program to circumvent this not just a mysql cli client e.g. `vim`, `less`, `tail -f`, etc) but as the only users are within the company (and they are not that crafty to be holding ssh connections open on purpose, more just lazy) and closing of the mysql cli on timeout is not available on the normal client it would ease my workload if (if I deployed mycli server wide) instead of having to force-acbly logoff users who have held a connection open all day, closed their laptop and went home (yet the server still thinks they are logged in using the long running process of the mysql client).

**Comment by amjith** [comment]
I see. Let me explain what it does and maybe that'll assuage your concerns. 

When the user first launches `mycli` it will establish a connection with the server and keep that connection alive as long as there is activity from the client. If the idle timeout has reached then the connection will be closed by the server and `mycli` won't try to reestablish the connection automatically. 

When the user comes back to mycli prompt and types in a query and press enter, it will tell the user that the connection is lost and ask their permission whether to reestablish the connection or not. 

To summarize, mycli will not attempt to reestablish connection with the server after an idle timeout. It will only connect back when the user explicitly asks for it. So the scenario that you've described should not be a concern. 

Let me know if that satisfies your need.

**Comment by tsroten** [comment]
@alasdairdc Are you worried about them keeping SSH connections open or MySQL connections open?

**Comment by alasdairdc** [comment]
@tsroten I'm more worried about the ssh connections staying open due to the mysql client not exiting automatically and thought this would be a feature we could add, say a config variable `exit_on_timeout=true/false` (defaulting to false of course) that exits `mysql` when the timeout is detected thus dumping back to the shell and allowing the shell timeouts to kick them off the ssh connection.

**Comment by tsroten** [comment]
@alasdairdc I can see what you mean. Users being lazy can certainly contribute negatively to your server's security. At the same time, I wonder if this is better handled in your SSH server config? There are options to handle SSH client idling, client timeout, etc. The problem you are describing is certainly not mycli-specific -- it's SSH-specific. Handling this on the SSH-side seems like the most sure way to achieve your desired results.

**Comment by tsroten** [comment]
@alasdairdc I do hope you'll be able to accomplish on the SSH server. I'm going to go ahead and close this issue. If you feel like there is more to be discussed here, feel free to open it back up.
